test/hotspot/jtreg/compiler/lib/verify/Verify.java

```diff
@@ -71,107 +109,176 @@ private static void checkEQ(Object a, Object b, String context) {
         if (ca != cb) {
             System.err.println("ERROR: Verify.checkEQ failed: class mismatch.");
             System.err.println("       " + ca.getName() + " vs " + cb.getName());
-            print(a, "a " + context);
-            print(b, "b " + context);
+            print(a, b, field, aParent, bParent);
             throw new VerifyException("Object class mismatch.");
         }
 
+        // Already visited? This makes sure that we are not stuck in cycles, and that we have
+        // a mapping of pairs (a, b) for structurally equivalent Objects.
+        if (checkAlreadyVisited(a, b, field, aParent, bParent)) {
+            return;
+        }
+
         switch (a) {
-            case Object[]  x -> checkEQimpl(x, (Object[])b,                context);
-            case Byte      x -> checkEQimpl(x, ((Byte)b).byteValue(),      context);
-            case Character x -> checkEQimpl(x, ((Character)b).charValue(), context);
-            case Short     x -> checkEQimpl(x, ((Short)b).shortValue(),    context);
-            case Integer   x -> checkEQimpl(x, ((Integer)b).intValue(),    context);
-            case Long      x -> checkEQimpl(x, ((Long)b).longValue(),      context);
-            case Float     x -> checkEQimpl(x, ((Float)b).floatValue(),    context);
-            case Double    x -> checkEQimpl(x, ((Double)b).doubleValue(),  context);
-            case byte[]    x -> checkEQimpl(x, (byte[])b,                  context);
-            case char[]    x -> checkEQimpl(x, (char[])b,                  context);
-            case short[]   x -> checkEQimpl(x, (short[])b,                 context);
-            case int[]     x -> checkEQimpl(x, (int[])b,                   context);
-            case long[]    x -> checkEQimpl(x, (long[])b,                  context);
-            case float[]   x -> checkEQimpl(x, (float[])b,                 context);
-            case double[]  x -> checkEQimpl(x, (double[])b,                context);
-            case MemorySegment x -> checkEQimpl(x, (MemorySegment) b,      context);
+            case Object[]  x -> checkEQimpl(x, (Object[])b,                 field, aParent, bParent);
+            case Byte      x -> checkEQimpl(x, ((Byte)b).byteValue(),       field, aParent, bParent);
```

Comment:
Can't you just pass `(Byte) b` to rely on auto unboxing instead?

Reply:
You are right, simplified it :)